src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -634,16 +682,24 @@ namespace AppInstaller::Repository
 
                 if (!otherComposite ||
                     static_cast<bool>(m_installedPackage) != static_cast<bool>(otherComposite->m_installedPackage) ||
-                    (m_installedPackage && !m_installedPackage->IsSame(otherComposite->m_installedPackage.get())) ||
                     m_availablePackages.size() != otherComposite->m_availablePackages.size())
                 {
                     return false;
                 }
 
+                if (m_installedPackage)
+                {
+                    if (m_installedPackage->GetSourceId() != otherComposite->m_installedPackage->GetSourceId() ||

```

Comment:
Looks like it will cause a crash if otherComposite->m_installedPackage  is nullptr?

Also nit: the 2 if statements could be combined

Reply:
nvm, I saw the `static_cast<bool>(m_installedPackage) != static_cast<bool>(otherComposite->m_installedPackage)` which ensures that.

But I think below `m_installedPackage->GetPackage()->IsSame(otherComposite->m_installedPackage->GetPackage().get())` already checked the sourceId is same, does not seem necessary to check explicitly here